quinn-proto/src/packet.rs

```diff
@@ -124,6 +125,7 @@ impl PartialDecode {
                     src_cid,
                     token,
                     number,
+                    version: Default::default(),
```

Comment:
since the same enum is used for encoding/decoding but the version is only needed for encoding this is a little uggly.

Reply:
Rather than making something up here and below, let's pipe the packet's actual version number through `PlainHeader` in each applicable case, here and below. Aside from being cleaner, this may come in handy when we implement version negotiation.

Reply:
Looks like you missed this one?